3.Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%? (Провести двусторонний тест.)

 - H_0: M = M_0
 - H_1: M != M_0
  n = 10
  m = 200
  доверительная вероятность равна 99%  a = 1% = 0,01
  так ка нужно провести двусторонний тест, то а = 0,01/02

In [5]:
import numpy as np

n = 10
m = 200
lol = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
X = np.mean(lol)
sigma = np.std(lol,ddof=1)

def N(sigma, X, m, n):
    n = 10
    m = 200
    return (X - m) / (sigma / np.sqrt(n))
t_n = N(sigma, X, m, n)
t_n

-1.0651074037450896

 
Т.к. распределение Стьюдента зависит от степеней свободы(n-1), мы строим 99% интервал, т.е.   а = 0,01/02 = 0,005
находим табличное значение t
t = 3.250 - табличное значение
t_n = -1.0651074037450896 - наблюдаемое зхначение
t > t_n   - табличное значение больше наблюдаемого значения, значит гипотеза верна 
принимаем нулевое значение
 -3.250<-1.0651074037450896<3.250

